clang/lib/Driver/Driver.cpp

```diff
@@ -1018,16 +1025,11 @@ void Driver::CreateOffloadingDeviceToolChains(Compilation &C,
     // If -fsycl is supplied without -fsycl-*targets we will assume SPIR-V
     // unless -fintelfpga is supplied, which uses SPIR-V with fpga AOT.
     // For -fsycl-device-only, we also setup the implied triple as needed.
-    StringRef SYCLTargetArch;
-    if (C.getInputArgs().hasArg(options::OPT_fsycl_device_only))
-      if (C.getDefaultToolChain().getTriple().getArch() == llvm::Triple::x86)
-        SYCLTargetArch = "spir";
-      else
-        SYCLTargetArch = "spir64";
-    else if (HasValidSYCLRuntime)
-      // Triple for -fintelfpga is spir64_fpga.
-      SYCLTargetArch = SYCLfpga ? "spir64_fpga" : "spir64";
-    if (!SYCLTargetArch.empty()) {
+    if (HasValidSYCLRuntime) {
+      StringRef SYCLTargetArch = getDefaultSYCLArch(C);
+      if (SYCLfpga)
+        // Triple for -fintelfpga is spir64_fpga.
+        SYCLTargetArch = "spir64_fpga";
```

Comment:
Just to clarify: this _does_ impact full compilation scenarios, right? In this case, does the E2E testing coverage that we have prove that E2E compilation/execution of DPC++ apps it's going to work on 32-bit systems? Or instead, could it be that it has never worked properly due to pointer size mismatch?

Reply:
In all honestly, I don't think we have done any testing on 32-bit systems.  All of our configurations are 64-bit hosted and the use of spir64 is commonplace.  The issue of the mismatch was only found due to the single internal 32-bit Windows build configuration (which isn't tested, it is just a build AFAIK)